File: src/snippets/bot-protection/reference/nestjs/Filtering.ts
import { botCategories, detectBot } from "@arcjet/nest";
// ...
// This is part of the rules constructed using withRule or a guard
// ...
detectBot({
  mode: "LIVE",
  // configured with a list of bots to allow from
  // https://arcjet.com/bot-list - all other detected bots will be blocked
  allow: [
    // filter a category to remove individual bots from our provided lists
    ...botCategories["CATEGORY:GOOGLE"].filter(
      (bot) => bot !== "GOOGLE_ADSBOT" && bot !== "GOOGLE_ADSBOT_MOBILE",
    ),
  ],
});

File: src/snippets/bot-protection/reference/python-fastapi/Filtering.py
import os

from arcjet import Mode, arcjet, detect_bot

# The Python SDK does not export a category-to-bots mapping. To allow a
# category but exclude specific bots, list the bot names you want to allow
# explicitly. See https://arcjet.com/bot-list for the full list.
aj = arcjet(
    key=os.environ["ARCJET_KEY"],
    rules=[
        detect_bot(
            mode=Mode.LIVE,
            # Allow most Google crawlers but block GOOGLE_ADSBOT and
            # GOOGLE_ADSBOT_MOBILE by listing the allowed bots individually.
            allow=[
                "GOOGLEBOT",
                "GOOGLEBOT_IMAGE",
                "GOOGLEBOT_NEWS",
                "GOOGLEBOT_VIDEO",
                "GOOGLE_FAVICON",
                "GOOGLE_INSPECTION_TOOL",
                "GOOGLE_SITE_VERIFICATION",
            ],
        ),
    ],
)
